src/main/scala/com/fortysevendeg/android/sqlite/resultset/SQLDroidResultSet.scala

```diff
@@ -0,0 +1,688 @@
+package com.fortysevendeg.android.sqlite.resultset
+
+import java.io._
+import java.math.BigDecimal
+import java.net.URL
+import java.sql._
+import java.text.SimpleDateFormat
+import java.util
+import java.util.Calendar
+
+import android.database.Cursor
+import com.fortysevendeg.android.sqlite._
+
+import scala.util.{Failure, Success, Try}
+import scala.{Array => SArray}
+
+class SQLDroidResultSet(
+  cursor: Cursor,
+  log: LogWrapper = new AndroidLogWrapper())
+  extends ResultSet
+  with NonEditableResultSet
+  with WrapperNotSupported {
+
+  private[this] var lastColumnRead: Option[Int] = None
+
+  private[this] def moveOp(f: => Boolean): Boolean =
+    f match {
+      case true =>
+        lastColumnRead = None
+        true
+      case false =>
+        false
+    }
+
+  private[this] def getOp[T](index: Int)(f: Int => T): T = {
```

Comment:
I've seen that this function is always using a method when the result returned is a false... Does it make sense if this _getOp_ method returns a Option[T]?

```
getOp(columnIndex.index)(cursor.isNull) match {
   case true => None
   case false => Option(f)
}
```
could replaced by:
```
getOp(columnIndex.index)(cursor.isNull)
```